src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -677,5 +677,39 @@ impl<E: Encoder> InvertedIndex<E> {
     }
 }
 
+/// A reader that filters out records that do not match a given field mask. It is used to
+/// filter records in an index based on their field mask, allowing only those that match the
+/// specified mask to be returned.
+pub struct FilterMaskReader<I> {
```

Comment:
Should we get rid of `DecoderResult` and have `Decoder::decoder()` returns `RSIndexResult` directly then?

Reply:
Yes, that change is happening in #6510 over here -> https://github.com/RediSearch/RediSearch/pull/6510/files#diff-2ee9bad447a86c8ba40c7a36e41c24038e3d75a710404be5e297d5bc34313fa4R556